client/state/application-passwords/reducer.js

```diff
@@ -0,0 +1,21 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { reject } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	APPLICATION_PASSWORD_DELETE_SUCCESS,
+	APPLICATION_PASSWORDS_RECEIVE,
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+
+export default createReducer( [], {
+	[ APPLICATION_PASSWORD_DELETE_SUCCESS ]: ( state, { appPasswordId } ) =>
+		reject( state, { ID: appPasswordId } ),
+	[ APPLICATION_PASSWORDS_RECEIVE ]: ( state, { appPasswords } ) => appPasswords,
+} );
```

Comment:
`createReducer` has some overhead and I don't think it adds any value over a function with a `switch` here. I'd like to discourage its use for simple reducers like this one.

Reply:
Good call - addressed in 3c2eb9927b2c8102ca16822a8f03efad67888f2f.